src/main/scala/modbat/dsl/Action.scala

```diff
@@ -49,18 +48,20 @@ class Action(val model: Model, val transfunc: () => Any, val method: Method = nu
 
   def maybeNextIf(conditions: (() => Boolean, String)*)
     (implicit line: sourcecode.Line, fullName: sourcecode.FullName): Action = {
-    for (cond <- conditions) {
-      nextStatePred += ((cond._1, new State(cond._2), true,
-                         ((fullName.value, line.value))))
+    conditions.foreach {
+      case (pred, stateName) =>
+        val nextStatePred = NextStatePred(pred, new State(stateName), maybe = true, fullName.value, line.value)
+        nextStatePreds += nextStatePred
     }
     this
   }
 
   def nextIf(conditions: (() => Boolean, String)*)
     (implicit line: sourcecode.Line, fullName: sourcecode.FullName): Action = {
-    for (cond <- conditions) {
-      nextStatePred += ((cond._1, new State(cond._2), false,
-                         ((fullName.value, line.value))))
+    conditions.foreach {
+      case (pred, stateName) =>
+        val nextStatePred = NextStatePred(pred, new State(stateName), maybe = true, fullName.value, line.value)
```

Comment:
here, you use `maybe = true`, perhaps copied from above; it should be `maybe = false`.
A few test cases fail, and I think this is the reason why.